File: trunk/ShadowFrameworkJS/src/shadow/system/data/objects/SFShortArray.js
//Java to JS on 15/03/2012

function SFShortArray(shortValues){
	this.shortValues=shortValues;
	//n is ignored
}

inherit(SFShortArray,SFPrimitiveType);

SFShortArray.prototype["getShortValues"]=function(){
			return this.shortValues;
		};

SFShortArray.prototype["readFromStream"]=function(stream){
			var  n = stream.readInt();
			this.shortValues  = stream.readShorts(n);
		};

SFShortArray.prototype["writeOnStream"]=function(stream){
			stream.writeInt(this.shortValues.length);
			stream.writeShorts(this.shortValues);
		};

SFShortArray.prototype["clone"]=function(){
			return new SFShortArray(this.shortValues.length);
		};

File: trunk/sfoglintegration/SF20Data/src/shadow/system/data/objects/SFShortArray.java
package shadow.system.data.objects;

import shadow.system.data.SFInputStream;
import shadow.system.data.SFOutputStream;
import shadow.system.data.SFWritableDataObject;

public class SFShortArray implements SFWritableDataObject{

	private short[] shortValues;
	
	public SFShortArray(short[] shortValues) {
		super();
		this.shortValues = shortValues;
	}

	public short[] getShortValues() {
		return shortValues;
	}

	@Override
	public void readFromStream(SFInputStream stream) {
		int n=stream.readInt();
		shortValues=stream.readShorts(n);
	}
	
	@Override
	public void writeOnStream(SFOutputStream stream) {
		stream.writeInt(shortValues.length);
		stream.writeShorts(shortValues);
	}
	
	@Override
	public SFShortArray copyDataObject(){
		return new SFShortArray(new short[shortValues.length]);
	}
	

	@Override
	public void setStringValue(String value) {
		shortValues=SFCharsetObjectUtils.readShorts(value, getClass().getSimpleName());
	}
	
	@Override
	public String toStringValue() {
		return SFCharsetObjectUtils.writeShorts(shortValues);
	}
}
